modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/SplitProcessor.java

```diff
@@ -55,12 +58,29 @@ String getSeparator() {
         return separator;
     }
 
+    boolean isIgnoreMissing() {
+        return ignoreMissing;
+    }
+
     @Override
     public void execute(IngestDocument document) {
-        String oldVal = document.getFieldValue(field, String.class);
-        if (oldVal == null) {
+        String oldVal;
+
+        try {
```

Comment:
Maybe we should replace this try-catch block with the a new helper method in `IngestDocument`?

```java
public <T> T getFieldValue(String path, Class<T> clazz, boolean ignoreMissing) {
        try {
            return getFieldValue(path, clazz);
        } catch (IllegalArgumentException e) {
            if (ignoreMissing) {
                return null;
            } else {
                throw e;
            }
        }
    }
```

Not sure about it, but it would save some code repetition in other processors too.